include/dsn/c/app_model.h

```diff
@@ -292,51 +292,54 @@ enum dsn_chkpt_apply_mode
     DSN_CHKPT_LEARN
 };
 
-typedef dsn_error_t (*dsn_app_checkpoint)(
-    void*,  ///< context from dsn_app_create
-    int64_t ///< checkpoint version
+typedef void(*dsn_app_on_batched_rpc_requests)(
+    void*,           ///< context from dsn_app_create
+    int64_t,         ///< ballot
+    int64_t,         ///< decree
+    dsn_message_t*,  ///< request array ptr
+    int              ///< request count
+    );
+
+typedef int(*dsn_app_get_internal_error)(
+    void*     ///< context from dsn_app_create
     );
 
-typedef dsn_error_t(*dsn_app_checkpoint_async)(
-    void*,  ///< context from dsn_app_create
-    int64_t ///< checkpoint version
+typedef dsn_error_t(*dsn_app_sync_checkpoint)(
```

Comment:
Well, I think easier-to-read (in this case, know what are the callbacks grouped into) is more important than "ugly". Anyway, this is a taste problem, and you may prefer what you like for now while we can do it consistently later after the merge.